lib/puppet/settings/http_extra_headers_setting.rb

```diff
@@ -0,0 +1,23 @@
+class Puppet::Settings::HttpExtraHeadersSetting < Puppet::Settings::BaseSetting
+
+  def type
+    :http_extra_headers
+  end
+
+  def munge(headers)
```

Comment:
I don't know how much we care about this, but this leaves open the possibility of a user not specifying headers correctly and not raising any errors.

For example, a user could supply a header, but no value that we can find. This will raise an error farther down the line

```
irb(main):016:0> [["string", "derp"], ["sadf"]].each do |name, value|
irb(main):017:1* puts name
irb(main):018:1> puts value
irb(main):019:1> headers[name] = value
irb(main):020:1> end
string
derp
TypeError: no implicit conversion of String into Integer
```

but it doesn't seem like a useful error at all.

```
irb(main):008:0> subject.munge('string')
=> [["string"]]
irb(main):009:0> subject.munge('string:derp')
=> [["string", "derp"]]
irb(main):010:0> subject.munge('string:derp,sadf:fasd')
=> [["string", "derp"], ["sadf", "fasd"]]
irb(main):011:0> subject.munge('string:derp,sadf')
=> [["string", "derp"], ["sadf"]]
```

Do we care? Should I make this better?

Reply:
It would be good to ensure every header has a non-empty value

Reply:
I'm validating this when we add custom headers to the headers hash rather than here